cms/articles/tests/test_series_models.py

```diff
@@ -99,14 +100,54 @@ def test_evergreen_route_related_data_renders_correctly(self):
         self.assertContains(response, f"All data related to {self.article_with_datasets.title}")
         self.assertContains(response, "Test dataset")
 
-    def test_evergreen_route_canonical_url(self):
+    def test_evergreen_route_related_data_canonical_url(self):
         """Test that the canonical URL on the related data page is the evergreen series URL."""
         response = self.client.get(f"{self.article_series_page.url}/related-data")
         self.assertEqual(response.status_code, HTTPStatus.OK)
         self.assertContains(
             response, f'<link rel="canonical" href="{self.article_series_page.get_full_url()}/related-data">', html=True
         )
 
+    def test_evergreen_route_related_data_latest_article_canonical_url(self):
+        """Test that latest article's related data page has evergreen canonical URL."""
+        # article_with_datasets is the latest article in setUp
+        response = self.client.get(f"{self.article_with_datasets.url}/related-data")
+        self.assertEqual(response.status_code, HTTPStatus.OK)
+        self.assertContains(
+            response, f'<link rel="canonical" href="{self.article_series_page.get_full_url()}/related-data">', html=True
+        )
+
+    def test_evergreen_route_related_data_non_latest_article_canonical_url(self):
+        """Test that non-latest article's related data page has its own canonical URL."""
+        # Create newer article to make the existing one non-latest
+        newer_article = StatisticalArticlePageFactory(
+            parent=self.article_series_page,
+            release_date=self.article_with_datasets.release_date + timezone.timedelta(days=1),
+        )
+        newer_article.datasets = StreamValue(
+            DatasetStoryBlock(),
+            stream_data=[
+                (
+                    "manual_link",
+                    {
+                        "title": "Newer dataset",
+                        "description": "Newer description",
+                        "url": "https://example.com/newer",
+                    },
+                )
+            ],
+        )
+        newer_article.save_revision().publish()
+
+        self.assertFalse(self.article_with_datasets.is_latest)
+
+        response = self.client.get(f"{self.article_with_datasets.url}/related-data")
+        self.assertEqual(response.status_code, HTTPStatus.OK)
+        # The canonical URL should be the article's own URL + /related-data (not the series)
+        # Check that it contains the article's edition URL path, not the series path
+        self.assertContains(response, f"{self.article_with_datasets.url}/related-data")
+        self.assertNotContains(response, f"{self.article_series_page.url}/related-data")
```

Comment:
Let's be more specific and check `rel="canonical" href`. This could pass because of other links, breadcrumbs, etc. that are already there. It doesn't protect against further extensions.